packages/yswaps/scripts/execute-fantom-pending-trades.ts

```diff
@@ -4,58 +4,125 @@ import uniswap from '@libraries/uniswap-v2';
-        hopTokensToTest: [SPOOKY_TOKEN, WFTM, WETH],
-      }));
-    } else if (compareAddresses(pendingTrade._swapper, spiritswapSwapper.address)) {
-      console.log(`Executing ${pendingTrade._id.toString()} through Spiritswap`);
-      ({ data } = await uniswap.getBestPathEncoded({
-        tokenIn: pendingTrade._tokenIn,
-        tokenOut: pendingTrade._tokenOut,
-        amountIn: pendingTrade._amountIn,
-        uniswapV2Router: SPIRITSWAP_ROUTER,
-        uniswapV2Factory: SPIRITSWAP_FACTORY,
-        hopTokensToTest: [SPIRIT_TOKEN, WFTM, WETH],
-      }));
+    const { data: spookyData, minAmountOut: spookyMinAmountOut } = await uniswap.getBestPathEncoded({
+      tokenIn: pendingTrade._tokenIn,
+      tokenOut: pendingTrade._tokenOut,
+      amountIn: pendingTrade._amountIn,
+      uniswapV2Router: SPOOKYSWAP_ROUTER,
+      uniswapV2Factory: SPOOKYSWAP_FACTORY,
+      hopTokensToTest: [SPOOKY_TOKEN, WFTM, WETH],
+      slippage: SLIPPAGE_PERCENTAGE,
+    });
+
+    tradesSetup.push({
+      swapper: (await ethers.getContract('AsyncSpookyswap')).address,
+      data: spookyData,
+      minAmountOut: spookyMinAmountOut,
+    });
+
+    console.log('spooky:', spookyMinAmountOut, spookyData);
+
+    const { data: spiritData, minAmountOut: spiritMinAmountOut } = await uniswap.getBestPathEncoded({
+      tokenIn: pendingTrade._tokenIn,
+      tokenOut: pendingTrade._tokenOut,
+      amountIn: pendingTrade._amountIn,
+      uniswapV2Router: SPIRITSWAP_ROUTER,
+      uniswapV2Factory: SPIRITSWAP_FACTORY,
+      hopTokensToTest: [SPIRIT_TOKEN, WFTM, WETH],
+      slippage: SLIPPAGE_PERCENTAGE,
+    });
+
+    tradesSetup.push({
+      swapper: (await ethers.getContract('AsyncSpiritswap')).address,
+      data: spiritData,
+      minAmountOut: spiritMinAmountOut,
+    });
+
+    console.log('spirit:', spiritMinAmountOut, spiritData);
+
+    const { data: zrxData, minAmountOut: zrxMinAmountOut } = await zrx.quote({
+      chainId: Number(chainId),
+      sellToken: pendingTrade._tokenIn,
+      buyToken: pendingTrade._tokenOut,
+      sellAmount: pendingTrade._amountIn,
+      slippagePercentage: SLIPPAGE_PERCENTAGE / 100,
+    });
+
+    tradesSetup.push({
+      swapper: (await ethers.getContract('ZRX')).address,
+      data: zrxData,
+      minAmountOut: zrxMinAmountOut,
+    });
+
+    console.log('zrx:', zrxMinAmountOut, zrxData);
+
+    let bestSetup: TradeSetup = tradesSetup[0];
+
+    for (let i = 1; i < tradesSetup.length; i++) {
+      if (tradesSetup[i].minAmountOut!.gt(bestSetup.minAmountOut!)) {
+        bestSetup = tradesSetup[i];
+      }
     }
-    await tradeFactory['execute(uint256,bytes)'](pendingTrade._id, data, { gasLimit: 8_000_000 });
-    await sleep(5000);
+
+    const tx = await tradeFactory['execute(uint256,address,uint256,bytes)'](
+      pendingTrade._id,
+      bestSetup.swapper,
+      bestSetup.minAmountOut!,
+      bestSetup.data,
+      {
+        gasLimit: 8_000_000,
+      }
```

Comment:
why is this hardcoded?

Reply:
AFAIR hardhat was never able to estimate gas correctly :face_palm: